cpp/include/cudf/io/types.hpp

```diff
@@ -166,6 +168,21 @@ struct host_buffer {
   host_buffer(const char* data, size_t size) : data(data), size(size) {}
 };
 
+/**
+ * @brief Returns `true` if the type is byte-like, meaning it is reasonable to pass as a pointer to
+ * bytes.
+ *
+ * @tparam T The representation type
+ * @return `true` if the type is considered a byte-like type
+ */
+template <typename T>
+constexpr inline auto is_byte_like_type()
+{
+  return cuda::std::is_same_v<T, int8_t> || cuda::std::is_same_v<T, char> ||
+         cuda::std::is_same_v<T, uint8_t> || cuda::std::is_same_v<T, unsigned char> ||
+         cuda::std::is_same_v<T, std::byte>;
+}
```

Comment:
Can this be replaced completely with `sizeof(T) == 1`?

Reply:
I've considered this as well. But, there's no guarantee that a one-byte object directly represents its data. If it's some sort of a pointer/index to actual data, we should not allow it to be treated as raw data. 
Maybe adding `is_integral` to `sizeof(T) == 1` would suffice?

Reply:
I'm not sure `is_integrate(std::byte)==true`.

Reply:
Anyway, if we don't have a perfect solution then keeping this check function is fine :smile:

Reply:
I'd love a perfect solution. The original plan was to accept `std::byte` only, but this produces problems with usage since it can't be implicitly cast. This template was added before we decided to deprecate the original API instead of remove it though. I could be convinced to only support `std::byte` on the new API, but it will require `reinterpret_cast` about everywhere it is used currently.

Reply:
I've spent so much time looking for the right type trait ;(
The best I can come up with is `is_trivial` plus sizeof() == 1.

I think I need to experiment a bit with how passing a std::string gets resolved if the filepath overload actually take a std::filepath object. That could impact the decision to support host_span<char>.

Reply:
Given that there may be uncertainty with the alternative solution, I'm fine with using the current one :smile: